src/distdl/utilities/slicing.py

```diff
@@ -103,3 +103,14 @@ def range_index(shape):
     # dimensions
     for x in itertools.product(*[range(y) for y in shape]):
         yield x
+
+
+def worker_layout(shape):
```

Comment:
Yes! You mentioned in a different comment that computing the storage workers this way will be too slow. I don't think there's a way to turn this into a generator in such a way that would prevent me from having to iterate over the entire thing, so I'll have to think about this again to efficiently compute the storage workers. I might get rid of this function altogether.

Reply:
I'm not sure it would be *too slow* since in general it should be small, but I want to avoid that if possible.  Iterating over the entire generator once shouldn't be too slow though.